### ISSUE TITLE
Augmentation of the validation dataset due to splitting...

### ISSUE THREAD

**Issue by kuzand** [udacity/deep-learning-v2-pytorch#132]
In the cifar-10 cnn notebook (and the mnist-mlp I think), the train dataset is split into train and validation datasets after the train dataset is augmented (with random rotations etc.). As result this data augmentation is transferred also to the validation dataset. A simpler way to do the split is by using the `torch.utils.data.random_split` function like this:
```
train_data = datasets.CIFAR10('data', train=True,
                              download=True, transform=train_transform)

test_data = datasets.CIFAR10('data', train=False,
                             download=True, transform=transforms.Compose([transforms.ToTensor(), 
                                                                          transforms.Normalize(mean, std)]))


train_size = int(0.8 * len(train_data))
valid_size = len(train_data) - train_size
train_dataset, valid_dataset = torch.utils.data.random_split(train_data, [train_size, valid_size])


train_loader = torch.utils.data.DataLoader(train_dataset, batch_size=batch_size,
                                           num_workers=num_workers)
valid_loader = torch.utils.data.DataLoader(valid_dataset, batch_size=batch_size,
                                           num_workers=num_workers)
test_loader = torch.utils.data.DataLoader(test_data, batch_size=batch_size, 
                                          num_workers=num_workers)
```
This also fixes another problem mentioned [here](https://github.com/udacity/deep-learning-v2-pytorch/issues/129), since now:
```
>>> print(len(train_loader.dataset))
>>> print(len(valid_loader.dataset))
40000
10000
```
However this fix will not work on Udacity workspace because the function `torch.utils.data.random_split` is available only in Pytorch versions >= 0.4.1, but Udacity still uses the old 4.0.0 version :)

**Comment by jaintj95** [comment]
@kuzand - So, from what I understand we do not want to pass augmented data from training set to the validation set because if we pass say - the same image which was rotated by 30 degrees during augmentation it will artificially inflate the validation set accuracy to an extent?
Also, we'll lose augmentations from training data? Am I understanding this correct?
Last question, how does your implementation avoid this? We are still performing the augmentations before splitting.

**Comment by abksyed** [comment]
@kuzand - As per your code the train_transform also gets applied to the validation set, which gives incorrect validation accuracy metric. Any workaround for this issue?

**Comment by abhiojha8** [comment]
Ideally, we should first split the dataset and then perform augmentation only on the training data. We will keep this feedback in mind while planning a refresh. Thanks :)
